pkg/errors/errors.go

```diff
@@ -124,3 +126,34 @@ func Cause(err error) error {
 
 	return err
 }
+
+// MultiError is an error that wraps multiple errors.
+type MultiError interface {
+	error
+	Unwrap() []error
+}
+
+// Join returns an error that wraps the given errors.
```

Comment:
> aggregate errors if there is clear use for the user, but not in general, i.e. fail fast is the default.

I strongly agree with this guidance you added in https://github.com/crossplane/crossplane/pull/4311. My concern with wrapping errors is that folks will reach for the tool too often. What do you think about including this recommendation here in the docstring?

Reply:
done